eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -1,38 +1,71 @@
 param(
+  # Please use the RepoOwner/RepoName format: e.g. Azure/azure-sdk-for-java
+  [Parameter(Mandatory = $true)]
+  $RepoNameWithOwner,
+  # Please use the RepoOwner/RepoName format: e.g. Azure/azure-sdk-for-java
   [Parameter(Mandatory = $true)]
   $RepoOwner,
+  # Please use the RepoOwner/RepoName format: e.g. Azure/azure-sdk-for-java
+  [Parameter(Mandatory = $true)]
+  $RepoName,
   # Use this if a pull request might have been opened from one repo against another.
   # E.g Pull request opened from azure-sdk/azure-sdk prBranch --> Azure/azure-sdk baseBranch
   $ForkRepoOwner,
   [Parameter(Mandatory = $true)]
-  $RepoName,
-  [Parameter(Mandatory = $true)]
-  $BranchPrefix,
+  $BranchPrefix,  
+  # Date format: e.g. Tuesday, April 12, 2022 1:36:02 PM. Allow to use other date format.
+  $LastCommitOlderThan=$env:LASTCOMMITDATEOVERRIDE,
   [Parameter(Mandatory = $true)]
   $AuthToken
 )
 
 . (Join-Path $PSScriptRoot common.ps1)
 
-LogDebug "Operating on Repo [ $RepoName ]"
+function GetCommitDate($commitUrl) {
+  try {
+    $commitResponse = Invoke-WebRequest $commitUrl
+    $commitData = ConvertFrom-Json $commitResponse
+    if (!$commitData.committer -or !$commitData.committer.date) {
+      LogDebug "No date returned from the commit sha. Skipping..."
+    }
+    return $commitData.committer.date
+  }
+  catch {
+    LogError "Failed to invoke $commitUrl with exception:`n$_"
+    exit 1
+  }
+}
+
+LogDebug "Operating on Repo [ $Repo ]"
+if (($Repo -split "/").Count -lt 2) {
+  LogError "The $Repo does not have right format. Please use RepoOwner/RepoName. E.g. Azure/azure-sdk-for-java. Your input: $Repo"
+  exit 1
+}
+
 try{
-  $branches = (Get-GitHubSourceReferences -RepoOwner $RepoOwner -RepoName $RepoName -Ref "heads/$BranchPrefix" -AuthToken $AuthToken).ref 
+  $responses = Get-GitHubSourceReferences -RepoNameWithOwner $RepoNameWithOwner -Ref "heads/$BranchPrefix" -AuthToken $AuthToken
 }
 catch {
   LogError "Get-GitHubSourceReferences failed with exception:`n$_"
   exit 1
 }
 
-foreach ($branch in $branches)
+foreach ($res in $responses)
 {
+  if (!$res -or !$res.ref) {
+    LogDebug "No branch returned from the branch prefix $BranchPrefix on $Repo. Skipping..."
+    continue
+  }
+  $branch = $res.ref
   try {
     $branchName = $branch.Replace("refs/heads/","")
     $head = "${RepoOwner}/${RepoName}:${branchName}"
     LogDebug "Operating on branch [ $branchName ]"
-    $pullRequests = Get-GitHubPullRequests -RepoOwner $RepoOwner -RepoName $RepoName -State "all" -Head $head -AuthToken $AuthToken
+    $pullRequests = Get-GitHubPullRequests -RepoNameWithOwner $RepoNameWithOwner -State "all" -Head $head -AuthToken $AuthToken
 
     if ($ForkRepoOwner)
```

Comment:
Lets delete this code block unless you find any usages.

Reply:
Done.